### PR TITLE
[Android] default filter list settings

### PR DESCRIPTION
**Pull request by tapanmodh**
Resolves https://github.com/brave/brave-browser/issues/26587

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:
1. Open settings
2. Go to Brave Shields & privacy
3. Go to Content Filtering
4. You can enable disable default filter list

### CODE REVIEW

**Review by mkarolin — APPROVED** [review]
`strings`++

**Review by deeppandya — APPROVED** [review]
LGTM

**Review by samartnik — APPROVED** [review]
lgtm

File: browser/ui/android/strings/android_brave_strings.grd [hunk 1]
```diff
@@ -512,6 +512,12 @@ This file contains all "about" strings.  It is set to NOT be translated, in tran
       <message name="IDS_FILTER_LISTS_TITLE" desc="Title for Filter Lists settings.">
         Filter Lists
       </message>
+      <message name="IDS_DEFAULT_FILTER_LISTS" desc="Title for Default filter lists.">
+        Default filter lists
+      </message>
+      <message name="IDS_DEFAULT_FILTER_LISTS_SUMMARY" desc="Summary text for Default filter lists.">
+        Additional popular community lists. Note that enabling too many filters will degrade browsing speeds.
+      </message>
```
Comment (antonok-edm) [inline browser/ui/android/strings/android_brave_strings.grd:520]: I'm concerned about calling these "default" filter lists, since they are generally _off_ by default. We already reserve that naming for the _on_-by-default lists (although we don't show it in the UI anywhere).

Historically we've called these ones "regional lists". Ever since `brave://adblock` was replaced by `brave://settings/shields/filters` on Desktop, they're just labeled "Filter lists".

My recommendation is to follow the naming on Desktop (`Filter lists`). The title of the overall settings page can also be renamed to match accordingly (`Content Filters`).

Comment (tapanmodh) [reply browser/ui/android/strings/android_brave_strings.grd:520]: @timchilds @deeppandya what do you think? Should we follow the desktop or ios?

Comment (deeppandya) [reply browser/ui/android/strings/android_brave_strings.grd:520]: I agree we should remove the default and just keep the filter list as a copy.

File: android/java/org/chromium/chrome/browser/shields/ContentFilteringAdapter.java [hunk 2]
```diff
@@ -139,61 +149,96 @@ public void onClick(View v) {
...
             customFilterListViewHolder.itemView.setOnClickListener(view -> {
-                if (holder.getAdapterPosition() == mCustomFilterLists.size() + 1) {
+                if (holder.getAdapterPosition() == mCustomFilterLists.size() + ONE_ITEM_SPACE) {
                     mBraveContentFileringListener.onAddCustomFiltering();
                 }
             });
+        } else if (holder instanceof DefaultFilterListViewHolder) {
+            DefaultFilterListViewHolder filterViewHolder = (DefaultFilterListViewHolder) holder;
+            Map<String, Value> storage =
+                    mDefaultFilterLists[position - mCustomFilterLists.size() - THREE_ITEMS_SPACE]
```
Comment (SergeyZhukovsky) [inline android/java/org/chromium/chrome/browser/shields/ContentFilteringAdapter.java:167]: let's add a check that array size is bigger than the requested item position

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Review by antonok-edm — APPROVED** [review]

**Comment by kjozwiak** [comment]
Verification **PASSED** on `Pixel 6` running `Android 14` using the following build(s):

```
Brave | 1.60.56 Chromium: 117.0.5938.92 (Official Build) canary (64-bit)
--- | ---
Revision | 3b7f9ca8094bbdada8e04aa8531f4bd9becbc25b
OS | Android 14; Build/UPB5.230623.009; 34; REL
```

Using the STR/Cases outlined via https://github.com/brave/brave-core/pull/19670#issue-1850212175 and the information outlined via https://github.com/brave/brave-browser/issues/26587, ensured the following:

* ensured that the default `filter list` is visible via `Settings` -> `Brave Shields & privacy` -> `Content Filtering`
  * checked using both `Dark` & `Light` themes as well
* disabled/enabled `Fanboy's Mobile Notifications` via `Settings` -> `Brave Shields & privacy` -> `Content Filtering`
  * ensured that the `Open with` modal appears within reddit.com once the filter is disabled (proving that filters are working)
  * ensured that the `Open with` modal is NOT appearing once `Fanboy's Mobile Notifications` is re-enabled
* compared the default `filter list` between `1.58.131` via `brave://adblock and `1.60.56` via `Content Filtering` settings
  * basically ensured that we're not missing any entries when we moved into the native implementation
* ensured that the settings persist after restarting (stay enabled/stay disabled etc..)
* ensured that enabling/disabling the default `filter list` via the toggles doesn't crash Brave

**`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** | **`Example`** 
----------------|----------------|----------------|----------------|----------------|----------------
Screenshot_20230925-161621 | Screenshot_20230925-161628 | Screenshot_20230925-161710 | Screenshot_20230925-162000 | Screenshot_20230925-161940 | Screenshot_20230925-163130
